lbrynet/daemon/Daemon.py

```diff
@@ -3082,29 +3073,245 @@ def jsonrpc_stream_availability(self, uri, search_timeout=None, blob_timeout=Non
+            --rename=<rename>               : (str) change name of existing account
+            --default                       : (bool) make this account the default
+
         Returns:
-            pos args
+            (map) new or updated account details
+
         """
-        out = (pos_arg, pos_args, pos_arg2, pos_arg3, a_arg, b_arg)
-        response = yield self._render_response(out)
-        defer.returnValue(response)
+        wallet = self.wallet.default_wallet
+        if create:
+            self.error_if_account_exists(account_name)
+            if single_key:
+                address_generator = {'name': SingleKey.name}
+            else:
+                address_generator = {
+                    'name': HierarchicalDeterministic.name,
+                    'receiving': {
+                        'gap': receiving_gap or 20,
+                        'maximum_uses_per_address': receiving_max_uses or 2},
+                    'change': {
+                        'gap': change_gap or 6,
+                        'maximum_uses_per_address': change_max_uses or 2}
+                }
+            ledger = self.wallet.get_or_create_ledger('lbc_mainnet')
+            if seed or private_key or public_key:
+                account = LBCAccount.from_dict(ledger, wallet, {
+                    'name': account_name,
+                    'seed': seed,
+                    'private_key': private_key,
+                    'public_key': public_key,
+                    'address_generator': address_generator
+                })
+            else:
+                account = LBCAccount.generate(
+                    ledger, wallet, account_name, address_generator)
+            wallet.save()
+        elif delete:
+            account = self.get_account_or_error('account_name', account_name)
+            wallet.accounts.remove(account)
+            wallet.save()
+            return "Account '{}' deleted.".format(account_name)
+        else:
+            change_made = False
+            account = self.get_account_or_error('account_name', account_name)
+            if rename is not None:
+                self.error_if_account_exists(rename)
+                account.name = rename
+                change_made = True
+            if account.receiving.name == HierarchicalDeterministic.name:
+                address_changes = {
+                    'change': {'gap': change_gap, 'maximum_uses_per_address': change_max_uses},
+                    'receiving': {'gap': receiving_gap, 'maximum_uses_per_address': receiving_max_uses},
+                }
+                for chain_name in address_changes:
+                    chain = getattr(account, chain_name)
+                    for attr, value in address_changes[chain_name].items():
+                        if value is not None:
+                            setattr(chain, attr, value)
+                            change_made = True
+            if change_made:
+                wallet.save()
+
+        if default:
```

Comment:
Might not be worth noting, but wallets recovered from seeds lose this kind of info.

Reply:
You're saying the seed isn't stored after restoring the account from a seed? Or did you mean something else, re: "this kind of info"?

Reply:
After reading the next comment I know I know what you meant by "this kind of info". You're talking about the sub-account in an HD chain. It's not an issue here because each Account is actually an independent root. When restoring from seed you're restoring a single Account and adding it to the list of other accounts also restored from a seed.

The reason for having a `default` account is that the current desktop app does not know that  wallet can have more than one account and also because it's convenient to have a default account which is used for payments while having other accounts as read-only or for other reasons.

Reply:
Maybe we're talking about the same thing, but given that HD accounts are integer indexed and your accounts are strings, I assume that you wanted _named_ accounts, such as 'checking', or 'saving' which maps to index 0 and 1 respectively (as an example).  Further, setting the 'saving' at index 1 as the default is well and good, but when recovering the wallet from seed (i.e. on a new machine) the names and default account mappings would be lost, no?

Reply:
Just to reiterate the `Account` in the code above is *not* the same as HD accounts. An `Account` represents a distinct HD root (completely unrelated to other `Account` roots). When you restore from a seed you are restoring a single `Account` only, unrelated to other accounts.

Reply:
Got it. thanks.